File: Qubit Server/dev/src/org/labrad/qubits/channeldata/TriggerDataTime.java
package org.labrad.qubits.channeldata;


public class TriggerDataTime extends TriggerDataBase {

  private boolean[] vals;

  public TriggerDataTime(boolean[] vals) {
    this.vals = vals;
  }

  public void checkLength(int expected) {
    LengthChecker.checkLengths(vals.length, expected);
  }

  @Override
  public boolean[] get() {
    return vals;
  }
}

File: QubitServer/src/main/java/org/labrad/qubits/channeldata/TriggerDataTime.scala
package org.labrad.qubits.channeldata

class TriggerDataTime(vals: Array[Boolean]) extends TriggerDataBase {

  def checkLength(expected: Int): Unit = {
    LengthChecker.checkLengths(vals.length, expected)
  }

  override def get(): Array[Boolean] = {
    vals
  }
}
